### PR TITLE
RPRBLND-2027: "Object selection" always fire rendering from 0 spp, there is no property changes

### PR DESCRIPTION
**Pull request by VascoPi**
### PURPOSE
 "Object selection" always fires rendering from 0 spp, there is no property changes.

### EFFECT OF CHANGE
Fixed Object selection starts viewport rendering update.

### TECHNICAL STEPS
Improved condition for object selection checking bpy.types.Scene update.
Fixed unnecessary updates of denoiser and upscaler.

### CODE REVIEW

File: src/rprblender/engine/engine.py [hunk 1]
```diff
@@ -128,10 +128,13 @@ def setup_image_filter(self, settings):
...
                 self._enable_image_filter(settings)
 
+            return True
+
         elif self.image_filter:
             self._disable_image_filter()
+            return True
 
-        return True
+        return False
```
Comment (bnagirniak) [inline src/rprblender/engine/engine.py:137]: Logic is not changed in this function, please revert.

File: src/rprblender/engine/engine.py [hunk 2]
```diff
@@ -392,10 +395,13 @@ def setup_upscale_filter(self, settings):
...
                 self._enable_upscale_filter(settings)
 
+            return True
+
         elif self.upscale_filter:
             self._disable_upscale_filter()
+            return True
 
-        return True
+        return False
```
Comment (bnagirniak) [inline src/rprblender/engine/engine.py:404]: Same here, logic is not changed in this function, please revert.

File: src/rprblender/engine/viewport_engine.py [hunk 3]
```diff
@@ -586,8 +586,10 @@ def sync_update(self, context, depsgraph):
             for mat in volume_domain_mat:
                 updates.append((mat, None, None))
 
-        if context.selected_objects != self.selected_objects and not updates:
-            # only a selection change
+        # only a selection change
+        if context.selected_objects != self.selected_objects \
+                and len(updates) == 1 and isinstance(updates[0][0], bpy.types.Scene)\
+                and updates[0][1] == updates[0][2] == False:
```
Comment (bnagirniak) [inline src/rprblender/engine/viewport_engine.py:592]: Do you mean here: `updates[0][1] == False and updates[0][2] == False` ?

**Review by bnagirniak — CHANGES_REQUESTED** [review]
Commented in code

File: src/rprblender/engine/engine.py (hunk 1 above)
Comment (VascoPi) [reply src/rprblender/engine/engine.py:137]: It's dedicated to the Outliner window which causes updates when clicking on an object two times.

File: src/rprblender/engine/viewport_engine.py (hunk 3 above)
Comment (VascoPi) [reply src/rprblender/engine/viewport_engine.py]: Sure, thanks.

**Review by DagerD — APPROVED** [review]
Tested. Works ok.

**Review by bnagirniak — APPROVED** [review]
Looks good

**Pull request merged**